sos/plugins/mssql.py

```diff
@@ -0,0 +1,94 @@
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import os
+from sos.plugins import Plugin, RedHatPlugin
+
+
+class MsSQL(Plugin, RedHatPlugin):
+    """Microsoft SQL Server on Linux
+    """
+
+    plugin_name = "mssql"
+    profiles = ('services',)
+    packages = ('mssql-server',)
+
+    option_list = [
+        ('mssql_conf', 'SQL Server configuration file.', '',
+         '/var/opt/mssql/mssql.conf')
+    ]
+
+    def setup(self):
+        mssql_conf = self.get_option('mssql_conf')
+
+        # Pick error log file from mssql_conf.
+        # Expecting the following format
+        # ```
+        # [filelocation]
+        # errorlogfile = /var/opt/mssql/log
+        # [sqlagent]
+        # errorlogfile = /var/opt/mssql/log/sqlagentstartup.log
+        # [network]
+        # kerberoskeytabfile = /var/opt/mssql/secrets/mssql.keytab
+        # ```
+        # Default values are described in the document:
+        # https://docs.microsoft.com/en-us/sql/linux
+        # /sql-server-linux-configure-mssql-conf
+        # ?view=sql-server-linux-2017#mssqlconf-format
+        section = ''
+        # default values
+        errorlogfile = '/var/opt/mssql/log'
+        sqlagent_errorlogfile = '/var/opt/mssql/log/sqlagentstartup.log'
+        kerberoskeytabfile = None
+        try:
+            for line in open(mssql_conf).read().splitlines():
+                if line.startswith('['):
+                    section = line
+                    continue
+                words = line.split('=')
+                if words[0].strip() == 'errorlogfile':
+                    if section == '[filelocation]':
+                        errorlogfile = words[1].strip()
+                    elif section == '[sqlagent]':
+                        sqlagent_errorlogfile = words[1].strip()
+                elif words[0].strip() == 'kerberoskeytabfile':
+                    if section == '[network]':
+                        kerberoskeytabfile = words[1].strip()
+        except IOError as ex:
+            self._log_error('Could not open conf file %s: %s' % (mssql_conf,
+                                                                 ex))
+            return
+
+        # Collect AD authentication configuratoin
+        keytab_err = ('keytab file is specfieid in mssql_conf'
+                      ' but not found in %s' % kerberoskeytabfile)
+        if kerberoskeytabfile is not None:
+            if os.path.isfile(kerberoskeytabfile):
+                self.add_cmd_output('ls -l %s' % kerberoskeytabfile)
+                self.add_cmd_output('klist -e -k %s' % kerberoskeytabfile)
+            else:
+                self._log_error(keytab_err)
+
+        # Expecting mssql_conf doesn't includeno sensitive information.
+        self.add_copy_spec([
+            mssql_conf,
+            errorlogfile + '/*',
+            sqlagent_errorlogfile
```

Comment:
This should only add all the log files if `--all-logs` is given. Please also implement size limiting for the log files only (no need for configuration). Running `git grep 'get_option.*log_size'` will give you plenty of examples to look at.